presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -370,6 +372,10 @@ public PlanOptimizers(
                         new CreatePartialTopN(),
                         new PushTopNThroughUnion())));
 
+        // add push down dereference expression
+        builder.add(new PushDownDereferenceExpression(metadata, sqlParser));
+        builder.add(new IterativeOptimizer(stats, statsCalculator, estimatedExchangesCostCalculator, ImmutableSet.of(new InlineDereferenceExpressions())));
```

Comment:
is there any point of having this added to all optimizers if you cannot take advantage of this in table scan operator today?

Maybe we could just add dead code until the moment you have all the pieces needed to prune unreferenced columns.

Reply:
I see. After this part, I will start working on part II and I will not merge this until that is finished. what do you think?